planutils/__init__.py

```diff
@@ -1,58 +1,120 @@
 
 import argparse, os
 
+from planutils import settings
+from planutils.package_installation import PACKAGES
+
 
 def setup():
 
-    assert not_setup_yet(), "Error: planutils is already setup. Remove ~/.planutils to reset (warning: all cached planners will be lost)."
+    if setup_done():
+        print("\nError: planutils is already setup. Setting up again will wipe all cached packages and settings.")
+        if input("  Proceed? [y/N] ").lower() in ['y', 'yes']:
+            os.system("rm -rf %s" % os.path.join(os.path.expanduser('~'), '.planutils'))
+        else:
+            return
+
+    CUR_DIR = os.path.dirname(os.path.abspath(__file__))
 
     print("\nCreating ~/.planutils...")
     os.mkdir(os.path.join(os.path.expanduser('~'), '.planutils'))
     os.mkdir(os.path.join(os.path.expanduser('~'), '.planutils', 'bin'))
-    os.mkdir(os.path.join(os.path.expanduser('~'), '.planutils', 'bin', 'images'))
+
+    settings.save({
+        'installed': []
+    })
+
+    os.symlink(os.path.join(CUR_DIR, 'packages'),
+               os.path.join(os.path.expanduser('~'), '.planutils', 'packages'))
 
     print("Adding bin folder to path (assuming ~/.bashrc exists)...")
-    os.system('echo \'export PATH="$HOME/.planutils/bin:$PATH"\' >> ~/.bashrc')
-
-    print("Installing planner scripts...")
-    from planutils.planner_installation import PLANNERS
-    for p in PLANNERS:
-        script  = "#!/bin/bash\n"
-        script += "echo\n"
-        script += "echo 'Planner not installed!'\n"
-        script += "read -p \"Download & install? [y/n] \" varchoice\n"
-        script += "if [ $varchoice == \"y\" ]\n"
-        script += "then\n"
-        script += "  planutils --install " + p + "\n"
-        script += "fi\n"
-        script += "echo"
-        with open(os.path.join(os.path.expanduser('~'), '.planutils', 'bin', p), 'w') as f:
-            f.write(script)
-        os.chmod(os.path.join(os.path.expanduser('~'), '.planutils', 'bin', p), 0o0755)
+    with open(os.path.join(os.path.expanduser('~'), '.bashrc'), "a+") as f:
+        f.write("export PLANUTILS_PREFIX=\"~/.planutils\"")
+        f.write("export PATH=\"$PLANUTILS_PREFIX/bin:$PATH\"")
+
+    print("Installing package scripts...")
+    for p in PACKAGES:
+        if PACKAGES[p]['runnable']:
+            script  = "#!/bin/bash\n"
+            script += "if $(planutils check-installed %s)\n" % p
+            script += "then\n"
+            script += "  ~/.planutils/packages/%s/run $@\n" % p
+            script += "else\n"
+            script += "  echo\n"
+            script += "  echo 'Package not installed!'\n"
+            script += "  read -r -p \"  Download & install? [Y/n] \" varchoice\n"
+            script += "  varchoice=${varchoice,,}\n" # tolower
+            script += "  if [[ \"$varchoice\" =~ ^(yes|y|)$ ]]\n"
```

Comment:
This doesn't match the default value in the prompt (`[Y/n]` indicates a default "yes" but confirming without typing "yes" will not enter this branch).